cmd/cluster-agent/api/server.go

```diff
@@ -103,7 +103,7 @@ func StopServer() {
 func validateToken(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		path := r.URL.String()
-		if strings.HasPrefix(path, "/api/v1/metadata/") && len(strings.Split(path, "/")) == 7 {
+		if strings.HasPrefix(path, "/api/v1/metadata/") && len(strings.Split(path, "/")) == 7 || path == "/version" {
```

Comment:
What are the challenges with the `version` endpoint not being authenticated?

Reply:
Can you add a quick test case to https://github.com/DataDog/datadog-agent/blob/master/cmd/cluster-agent/api/server_test.go#L42. 

This just adds a sanity check in case someone changes the endpoint or the middleware.